tools/wptrunner/wptrunner/wpttest.py

```diff
@@ -313,32 +313,28 @@ class TestharnessTest(Test):
 
     def __init__(self, tests_root, url, inherit_metadata, test_metadata,
                  timeout=None, path=None, protocol="http", testdriver=False,
-                 jsshell=False, scripts=None):
+                 jsshell=False):
         Test.__init__(self, tests_root, url, inherit_metadata, test_metadata, timeout,
                       path, protocol)
 
         self.testdriver = testdriver
         self.jsshell = jsshell
-        self.scripts = scripts or []
```

Comment:
I do need *some* way to get these; they're used in https://searchfox.org/mozilla-central/rev/cb7faaf6b4ad2528390186f1ce64618dea71031e/js/src/tests/jstests.py#404

Reply:
@jgraham told me it was unused! :)

Guess we need to add it to the JSON, then.

Reply:
Well it's not that expensive to read from the file. But if the goal is that we don't read any test file data except metadata when the test file hasn't been updated, then yes.

Reply:
> But if the goal is that we don't read any test file data except metadata when the test file hasn't been updated, then yes.

We shouldn't need to read any test file data to just run the test.

Reply:
So I've fixed this by putting it in the manifest… but I fear this means we're now gonna be broken until the MANIFEST is regenerated from scratch given it won't contain this until it's regenerated…

Reply:
We can bump the manifest version number to force it to be regenerated. We should probably also make sure that all the manifest upload things regenerate from scratch every time.

Reply:
Following on from discussion on IRC, I've re-added `tests_root` to items (yay memory consumption!) and made it possible for them to create `SourceFile` objects (albeit behind a private property) to fetch the `scripts` from there if the `_extras` doesn't already contain it.

Reply:
And, to be clear, when we next bump the MANIFEST version we should eliminate all of that.